### ISSUE TITLE
Twig 3 Compatibility

### ISSUE THREAD

**Issue by toxpal** [nochso/html-compress-twig#15]
Hi @nochso 

I'm using your html-compress-twig for a few years and like it very much. I see it doesn't get much updates, but it works well, so it's fine :)

However, I noticed it doesn't work with Twig 3, which was released recently. Do you have any plans to support it? If not, maybe you can refer to some other maintained plugin which does that?

Thank you.

**Comment by StasToken** [comment]
I am also interested in this question, I spent all day today trying to find something similar on the Internet, but not a single analogue can twig 3

**Comment by antiftw** [comment]
Hi @nochso 

Bump, since I've been having the same problem migrating my cms to Symfony 5, which uses twig 3.0 ;) Have spent multiple days now looking for something with similar functionality and that also works without much hassle, without any success.

Thanks in advance,

PS. I don't even mind doing so myself, since I have the idea there are no major changes required, just a version bump, or am I wrong and are you dreading the day to start the rewrite?? haha ^^ Have never worked with submitting anything myself to packagist/github so you would have to give me some pointers where to start ;)

**Comment by ekyazilim** [comment]
Hi @nochso
I'm eagerly awaiting the upgrade to twig 3. thanks..

**Comment by MarelPup** [comment]
Hi @nochso 
the repo is silent since 2017 , 
is our best option implementing the dependency 
https://packagist.org/packages/wyrihaximus/html-compress
using kernel.response as such ? 
https://stackoverflow.com/a/72082233
https://symfony.com/doc/current/event_dispatcher.html#before-filters-with-the-kernel-controller-event

**Comment by toxpal** [comment]
Yes, I switched to the solution mentioned above (wyrihaximus/html-compress) earlier, but it doesn't minify inline JavaScript code...

**Comment by MarelPup** [comment]
strange, considering 
```php
    public function __construct($forceCompression = false)
    {
        $this->forceCompression = $forceCompression;
        $this->parser = Factory::constructSmallest();
        $this->callable = array($this, 'compress');
    }

    public function compress(Twig_Environment $twig, $html)
    {
        if (!$twig->isDebug() || $this->forceCompression) {
            return $this->parser->compress($html);
        }
        return $html;
    }
``` 
doesn't seem to be a different implemetation of wyrihaximus ' basic compression

**Comment by toxpal** [comment]
Sorry, my bad, I used it earlier (can't remember the reason, but I had to look for another solution), and switched to https://packagist.org/packages/voku/html-compress-twig later

**Comment by MarelPup** [comment]
It's ok, i had already moved to a homemade solution when writing my first comment , i think i'll use this solution

**Comment by toxpal** [comment]
If I understand right, you use wyrihaximus/html-compress as a base + some custom code?

**Comment by MarelPup** [comment]
yes, I use kernel.response , as cited in the stackoverflow link
